zerver/lib/user_groups.py

```diff
@@ -184,6 +185,17 @@ def access_user_group_for_update(
         NamedUserGroup.GROUP_PERMISSION_SETTINGS[permission_setting],
     )
 
+    # Users with permission to manage the group should be able to add members
+    # to the group. This also takes care of the case where a user creating a group
+    # with themselves having the permission to manage it don't have to explicitly
+    # add themselves to can_add_members_group.
+    if not user_has_permission and permission_setting == "can_add_members_group":
```

Comment:
This check may want to be `and permission_setting != "can_manage_group"`? I guess we can wait for adding code for future settings to help us decide what is cleaner.

Reply:
I've added the condition mentioned in the first sentence, for the second point, I agree, maybe that could be a good immediate followup for this PR?

Reply:
Yeah, second point can e a follow-up for sure, hence the "ultimately" phrasing.